File: sorting/quicksort.py
def quick_sort(lst):
        """Quick sort."""
        if len(lst) <= 1:
            return lst

        pivot = lst[0]
        left = []
        right = []

        for i in lst[1:]:

            if i < pivot:
                left.append(i)

            elif i > pivot:
                right.append(i)

        left = quick_sort(left)
        right = quick_sort(right)

        return left + [pivot] + right


if __name__ == '__main__':

    import timeit
    setup = 'from quicksort import quick_sort'

    print(min(timeit.Timer('quick_sort([10, 5, 7, 8, 12])', setup=setup).repeat(7, 1000))) 

File: js/quick_sort.js
//quick sort method.

var quick_sort = function(unsorted){

    if(unsorted.length <= 1){
        return unsorted
    }

    let pivot = unsorted.pop()
    let left = []
    let right = []
    let sorted = []

    for(let i = 0; i < unsorted.length; i++){
        if(unsorted[i] <= pivot){
            left.push(unsorted[i])
        }else if(unsorted[i] > pivot){
            right.push(unsorted[i])
        }
    }
    return sorted.concat(quick_sort(left), pivot, quick_sort(right));
}

unsorted = [10,7,4,200,6,8,5,28]


console.log(quick_sort(unsorted))

module.exports = quick_sort;
